implementations/php-json-schema/bowtieJsonSchema.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+require_once 'vendor/autoload.php';
+
+use Opis\JsonSchema\{
+    Validator
+};
+
+$STARTED = false;
+
+$validator = new Validator();
+$resolver = $validator->loader()->resolver();
+
+function start($request)
+{
+    if ($request->version !== 1) {
+        throw new Exception('Wrong version!');
+    }
+
+    $GLOBALS['STARTED'] = true;
+
+    $response = [
+        'ready' => true,
+        'version' => 1,
+        'implementation' => [
+            'language' => 'php',
+            'name' => 'json-schema',
+            'homepage' => 'https://opis.io/json-schema',
+            'issues' => 'https://github.com/opis/json-schema/issues',
+            'dialects' => [
+                'http://json-schema.org/draft/2020-12/schema#',
+                'http://json-schema.org/draft-2019-09/schema#',
+                'http://json-schema.org/draft-07/schema#',
+                'http://json-schema.org/draft-06/schema#',
+            ],
+            'os' => PHP_OS,
+            'os_version' => php_uname('r'),
+            'language_version' => PHP_VERSION,
+        ],
+    ];
+    return json_encode($response);
+}
+
+function dialect($request)
+{
+    if (!$GLOBALS['STARTED']) {
+        throw new Exception('Not started!');
+    }
+    return json_encode(['ok' => true]);
+}
+
+function run($request)
+{
+    if (!$GLOBALS['STARTED']) {
+        throw new Exception('Not started!');
+    }
+
+    $registry = json_decode($request->case->registry);
+
+    if ($registry != null) {
+        foreach ($registry as $key => $value) {
+            $GLOBALS['resolver']->registerRaw($value, $key);
```

Comment:
This shouldn't be global -- you're leaking the registry from one test to the next. Every test should run with a clean slate (no previously seen URIs), otherwise you may accidentally pass or fail tests based on previous state.

I'm guessing this means you should be re-instantiating `Validator` for each test case.

Reply:
Got it!